src/variable.cpp

```diff
@@ -3931,6 +3955,39 @@ unsigned int Variable::data_mask(char *str)
   return datamask;
 }
 
+
+/* ----------------------------------------------------------------------
+   generates a random prime number between 10'000 and INT_MAX
+
+   rand() is not the best, a more optimal soultion would be:
+
+      #define FIRTS_PRIME 10007     // first prime greather than 10'000
+      #define LAST_PRIME 2147483647 // last prime less than INT_MAX
+      std::random_device dev;
+      std::mt19937 rng(dev());
+      std::uniform_int_distribution<std::mt19937::result_type> dist(FIRTS_PRIME, LAST_PRIME);
+      seed = dist(rng);
+
+    but it depends on C++11, so the backward compatibility would be lost
+------------------------------------------------------------------------- */
+int Variable::random_seed(){
+  int seed = 0;
+
+  static const int FIRTS_PRIME = 10007;
+  static const int LAST_PRIME = 2147483647;
+  static const int SEED_RANGE = (LAST_PRIME - FIRTS_PRIME) + 1;
+
+  if(me == 0){
+    while(!MathExtraLiggghts::isPrime(seed)){
```

Comment:
This algorithm is really bad in terms of performance. Using lookup tables would be much better.

Reply:
Agreed, I'm afraid a list would be too big to hold it in the memory (105'097'397 numbers). I'll try to come up with a faster solution.

Reply:
Thanks! I'll look into that.